lib/sync/post.rb

```
@@ -0,0 +1,151 @@
+
+    def add_media_asset(asset_name, media_content)
+      storage.add_media_asset(slug, asset_name, media_content)
+    end
+
+    private
+
+    def assemble_content
+      "#{metadata.to_yaml(line_width: -1)}---\n#{body_markdown}"
+    end
+
+    def self.prepare_markdown(markdown)
```

RuboCop (1.69.1)

[warning] `private` (on line 114) does not make singleton methods private. Use `private_class_method` or `private` inside a `class << self` block instead.

(Lint/IneffectiveAccessModifier)